projects/plugins/jetpack/extensions/shared/external-media/sources/jetpack-app-media.js

```diff
@@ -0,0 +1,113 @@
+import { QRCode } from '@automattic/jetpack-components';
+import { useCallback, useEffect, useState } from '@wordpress/element';
+import { __, sprintf, _n } from '@wordpress/i18n';
+import { JetpackAppIcon } from '../../icons';
+import useInterval from '../../use-interval';
+import MediaBrowser from '../media-browser';
+import { MediaSource } from '../media-service/types';
+import withMedia from './with-media';
+
+function JetpackAppMedia( props ) {
+	const { media, insertMedia, isCopying, multiple, getMedia } = props;
+
+	const wpcomBlogId = window?.Jetpack_Editor_Initial_State?.wpcomBlogId || 0;
```

Comment:
Should we bail entirely if we do not have a WordPress.com blog ID, since we will not be able to do anything with `0`?

Reply:
Good call!

Reply:
because of this `isCurrentUserConnected()` call in https://github.com/Automattic/jetpack/pull/34179/files/8bcaa103d32508623c2bbfb7c190fde1e054c45e#diff-7aaabf6ea20f69c89116cad6960da44b10dca2ebdb72a2560dcf4d76e55e88c6R29 we should always have this be available. I don't think we need to explicitly return early here. 

Since this would be a weird case where the button would show but not the UI.